File: Backend/EmployeeManagementSystem/EmployeeManagementSystem.Domain/Entities/Employee.cs


using EmployeeManagementSystem.Domain.Enum;

namespace EmployeeManagementSystem.Domain.Entities
{
    public class Employee
    {
        public int Id { get; set; }
        public string FirstName { get; set; }  = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public Position Position { get; set; }
        public DateTime HireDate { get; set; }
        public Status Status { get; set; }

        public int DepartmentId { get; set; }
        public Department? Department { get; set; }

        public ICollection<Salary> Salaries { get; set; } = new List<Salary>();

    }
}

File: Frontend/frontend-emp-system/src/Types/employee.ts
import type { Department } from "./department";
import type { Position, Status } from "./enums";

export interface Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  position: Position;
  hireDate: string; // ISO date string (from DateTime in C#)
  status: Status;
  departmentId: number;
  department?: Department; // nullable in C#
}

export interface EmployeeResponse{
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  position: string;
  hireDate: string; // ISO date string (from DateTime in C#)
  status: string;
  departmentName: string;
  departmentId: number;
}

 export interface EmployeeRequest{

      firstName:string,
      lastName :string,
      email:string,
      phoneNumber:string,
      position :string,
      status :string,
      hireDate:string,
      departmentId :string
 }
